src/plugins/cesium/ngv-cesium-factories.ts

```diff
@@ -93,18 +95,36 @@ export async function instantiate3dTileset(
     const key = extraOptions?.key as string | undefined;
     return createGooglePhotorealistic3DTileset(key);
   }
-  if (typeof url === 'string') {
-    return Cesium3DTileset.fromUrl(
-      // This allows for offline mode
-      new Resource(url),
-      withExtra(config.options, extraOptions),
-    );
-  } else {
-    return Cesium3DTileset.fromIonAssetId(
-      url,
-      withExtra(config.options, extraOptions),
-    );
+
+  // Replaces Cesium ION id with cesium API url to make it work offline
+  // fixme
+  //  * When the page is loaded offline and then switched to online, the token will be missed and 3d tiles from ION will not work (except tiles loaded offline).
+  //  * The token expires and should be replaced after some period of time
+  let accessToken: string | undefined;
+  if (typeof url === 'number') {
+    accessToken = await getIonAssetToken(url);
+    url = `${ION_ASSETS_URL}${url}/tileset.json`;
   }
+  return Cesium3DTileset.fromUrl(
+    // This allows for offline mode
+    new Resource({
+      url,
+      headers: accessToken ? {Authorization: `Bearer ${accessToken}`} : {},
+    }),
```

Comment:
I agree we should use `fromUrl`.
But why not instantiate a IonResource?
Also, the accessToken should be received as context, (or you should have defined Ion.defaultAccessToken, not here but at app initialization and it will be used automatically by Cesium).

Reply:
> But why not instantiate a IonResource?
Also, the accessToken should be received as context, (or you should have defined Ion.defaultAccessToken, not here but at app initialization and it will be used automatically by Cesium).

it is not going to work in offline mode

Reply:
if we use IonResource with defaultAccessToken, the layer will not be initialized without an internet connection because it requires an additional request to get the layer token (different from the one we have)